core/src/test/java/tech/tablesaw/table/TableTransposeTest.java

```diff
@@ -0,0 +1,288 @@
+package tech.tablesaw.table;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+import tech.tablesaw.api.BooleanColumn;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.FloatColumn;
+import tech.tablesaw.api.IntColumn;
+import tech.tablesaw.api.LongColumn;
+import tech.tablesaw.api.StringColumn;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.columns.Column;
+import tech.tablesaw.columns.numbers.DoubleColumnType;
+
+public class TableTransposeTest {
+  private static final String TABLE_NAME = "Data";
+
+  @Test
+  void transposeEmptyTable() {
+    Table empty = Table.create(TABLE_NAME);
+    Table result = empty.transpose();
+    assertEquals(empty.print(), result.transpose().print());
+    assertTableEquals(TABLE_NAME, new Object[][] {}, result);
+  }
+
+  @Test
+  void transposeDoubles() {
+    Table testTable =
+        Table.create(
+            TABLE_NAME,
+            DoubleColumn.create("value1", new double[] {1.0, 1.1, 1.2}),
+            DoubleColumn.create("value2", new double[] {2.0, 2.1, 2.2}));
+    Table result = testTable.transpose();
+
+    assertTableEquals(
+        TABLE_NAME,
+        new Object[][] {
+          {"0", 1.0, 2.0},
+          {"1", 1.1, 2.1},
+          {"2", 1.2, 2.2},
+        },
+        result);
+  }
+
+  @Test
+  void transposeWithMissingData() {
+    Table testTable =
+        Table.create(
+            TABLE_NAME,
+            DoubleColumn.create(
+                "value1", new double[] {1.0, DoubleColumnType.missingValueIndicator(), 1.2}),
+            DoubleColumn.create("value2", new double[] {2.0, 2.1, 2.2}));
+    Table result = testTable.transpose();
+
+    assertTableEquals(
+        TABLE_NAME,
+        new Object[][] {
+          {"0", 1.0, 2.0},
+          {"1", DoubleColumnType.missingValueIndicator(), 2.1},
+          {"2", 1.2, 2.2},
+        },
+        result);
+  }
+
+  @Test
+  void transposeFloats() {
+    Table testTable =
+        Table.create(
+            TABLE_NAME,
+            StringColumn.create("label", new String[] {"row1", "row2", "row3"}),
+            FloatColumn.create("value1", new float[] {1.0f, 1.1f, 1.2f}),
+            FloatColumn.create("value2", new float[] {2.0f, 2.1f, 2.2f}));
+    assertEquals(
+        testTable.print(),
```

Comment:
Can we call `assertTableEquals` instead of checking the output of `print`?

Reply:
They are floating point numbers so equality is tricky, they need to be compared within a tolerance.
Print does some formatting which helps avoid that which is why this ended up like this.

Reply:
Have changed assertions to address this and removed the prints. I pulled to float values out into variables, which should allow them to be compared safely.